.github/workflows/check-cli-documentation.yml

```diff
@@ -0,0 +1,59 @@
+# Copyright 2024 Democratized Data Foundation
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.txt.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+# This workflow checks that all CLI documentation is up to date.
+# If the documentation is not up to date then this action will fail.
+name: Check CLI Documentation Workflow
+
+on:
+  pull_request:
+    branches:
+      - master
+      - develop
+
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+    branches:
+      - master
+      - develop
+
+jobs:
+  check-cli-documentation:
+    name: Check cli documentation job
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code into the directory
+        uses: actions/checkout@v3
+
+      - name: Check no changes exist initially
```

Comment:
question: Why is this step needed?

todo: If keeping, please document why we want this :)

Reply:
Incase upon checkout some dirty state exists (i don't think it's going to happen, but just for extra caution). It's be nice to catch that inconsistency.

Happy to remove if there is preference/consensus.

Reply:
Okay, I'm somewhat sceptical if a check here is the right place for that, but it is a cheap action and not worth spending time on :)

Please do document this though before merge.